### ISSUE TITLE
Migrate to react-leaflet v3

### ISSUE THREAD

**Issue by MasterEric** [clementallen/react-leaflet-textpath#48]
[The latest version of react-leaflet](https://github.com/PaulLeCam/react-leaflet) introduces many breaking changes and improvements, but this means the current version of react-leaflet-textpath is incompatibile with it.

**Comment by clementallen** [comment]
Hey @MasterEric - yes I'm aware of the new V3 library and have played around with it a little myself but not got round to upgrading this library to support it. If you feel like giving it a go that would be hugely appreciated.

I am keen to keep v2 support for those using it though as I'm sure many people (myself included) won't be able to easily upgrade to v3 in the near future as it requires a lot of changes.

Perhaps something along these lines for imports?  Having the new V3 as the default but keeping a named export for V2 for those who need it.

```javascript
// Default for react-leaflet v3
import TextPath from 'react-leaflet-textpath';

// Keep an option available for those still on v2
import { TextPathV2 } from 'react-leaflet-textpath';
```

**Comment by MasterEric** [comment]
For backwards compatibility, I recommend changing the library version on
this module to version 3.0. This will do two things:

1. It will indicate to users that this version corresponds with the 3.0
version of react-leaflet.

2. It will prevent breaking changes to existing projects. NPM will not auto
update though major versions, and old versions of the library are still
hosted, so you can simply tell users to install the 1.0 version of the
library to maintain react-leaflet v2 support.

I have personally been working on the required changes already in my own
fork, but I'm having issues getting it working. One problem I'm
experiencing is an issue where the test cannot run because React.useRef
cannot be emulated by enzyme with it's current configuration.

**Comment by clementallen** [comment]
I was always going to do a major version bump with these changes, and like the idea of moving to `3.0.0` in line with react-leaflet.

The reason I want to still support v2 in the library is if there are any new versions of the underlying leaflet textpath package, it can be bumped for both v2 & v3 without having to maintain a separate branch and releasing of two versions.  This approach seems to have worked well for react-leaflet plugins that supported both the old v1 & v2 versions some time ago.

Huge thanks for forking and trying out the upgrades!

**Comment by MasterEric** [comment]
I don't know what the convention for this is. You could see how other apps do it. I would suggest simply maintaining the 1.0 version of the library on a separate major version number. For example, if leaflet-textpath updates, you could release a `1.1.0` update and a `3.1.0` update to the library at the same time. Again, not sure how best to handle this use case, I recommend doing some research on how other libraries handle it.

Also watch out for my pull request soon, I'm doing the finishing touches on it now.

**Comment by clementallen** [comment]
Although my preference would be to keep they both supported in the same version, I've gone through all the other published v3 libraries and they've branched off the v2 support so I'll do the same to keep it consistent.

Look forward to seeing the pull request! 👍

**Comment by clementallen** [comment]
@MasterEric thanks for all your help, I've now released this as version `2.0.0`.

The bug in react-leaflet is still there unfortunately, and given previous observations of other issues raised over the years the maintainer is very unlikely to be open to any changes despite the current illogical behaviour.  Oh well 🙃
